src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubectlBuildWrapper.java

```diff
@@ -89,6 +72,26 @@ public String getCaCertificate() {
         return caCertificate;
     }
 
+    private String buildKubeConfig(KubernetesAuth auth) throws JsonProcessingException {
```

Comment:
Much cleaner using the API than building the cli args manually. 👍

Reply:
This method could be added to `KubernetesAuth` interface, implemented as this body by all subclasses except `KubernetesAuthKubeconfig`.